### DISCUSSION TITLE
Question: Search in Text Channels

### DISCUSSION THREAD

**Discussion by Ryan-FAA-PTCRoAV** [Pycord-Development/pycord#2519] [Q&A]
Is there a way to search inside a `discord.TextChannel'? It would be the same as searching on Discord with the parameter "in:channelname".

TextChannel Docs: https://docs.pycord.dev/en/stable/api/models.html#discord.TextChannel

**Comment by Lulalaby** [comment]
the search api is not for bots

**Reply by Ryan-FAA-PTCRoAV** [reply]
So, there is no possible way?

**Reply by Ryan-FAA-PTCRoAV** [reply]
channel.history might work.

**Reply by NeloBlivion** [reply]
You can already filter channel.history yourself, *maybe* we could add a `check` where you pass in a function but the general point is that no message search api is exposed to bots.
